.codesandbox/templates/next/components/index.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { createRef, useEffect, useMemo } from 'react';
+import {
+  DOMManager,
+  DOMManagerType,
+  StaticCanvas,
+  StaticDOMManagerType,
+} from 'fabric';
+
+const canvasStyle: React.CSSProperties = {
+  position: 'absolute',
+  left: 0,
+  top: 0,
+  touchAction: 'none',
+  msTouchAction: 'none',
+  userSelect: 'none',
+};
+
+export const CanvasSlot = ({ name }: { name: string }) => <div name={name} />;
+
+/**
+ * @example Standard
+ *   const create = useCallback((arg0: DOMManagerType) => {
+ *     const canvas = new Canvas(arg0);
+ *     ...
+ *     return canvas; // return for cleanup
+ *   });
+ *
+ *   <CanvasComponent create={create}>
+ *     <CanvasSlot name="main" />
+ *     <CanvasSlot name="top" />
+ *   </CanvasComponent>
+ *
+ * @example DOM elements in between
+ *   <CanvasComponent create={create}>
+ *     <CanvasSlot name="main" />
+ *     <CanvasSlot name="pointers" />
+ *     Hello World
+ *     <CanvasSlot name="top" />
+ *   </CanvasComponent>
```

Comment:
@jiayihu I believe you were looking for this

Reply:
Not really, we create the rendering layers in the fabric code, so imperately, not declaratively in React. But it's definitely useful to have simplified APIs to create other rendering layers in fabric. For instance we might want to move rendering controls to its own layer for performance reasons, e.g. to support hovering controls without having to re-render everything.